src/main/kotlin/com/sourcegraph/cody/ui/TransparentButton.kt

```diff
@@ -1,16 +1,19 @@
 package com.sourcegraph.cody.ui
 
 import com.intellij.ide.ui.UISettings
+import com.intellij.ui.ColorUtil
 import com.intellij.util.ui.UIUtil
 import java.awt.*
 import javax.swing.JButton
 
 class TransparentButton(text: String) : JButton(text) {
-  private val opacity = 0.7f
+  private val enabledOpacity = 0.7f
+  private val disabledOpacity = 0.4f
   private val cornerRadius = 5
   private val horizontalPadding = 10
   private val verticalPadding = 5
-  private val textColor = UIUtil.getLabelForeground()
+  private val enabledTextColor = UIUtil.getLabelForeground()
+  private val disabledTextColor = ColorUtil.darker(UIUtil.getLabelForeground(), 3)
```

Comment:
I think most of these variables can be locally defined in function bodies. I think it's unnecessary to place them in the class field if we only use them once. It needlessly clutters the code, unless we're doing some optimizations here - please correct me if yes.